mirrord/layer/src/debugger_ports.rs

```diff
@@ -91,81 +111,162 @@ impl FromStr for DebuggerType {
+                .into_iter()
+                .collect::<Vec<_>>()
             }
             Self::JavaAgent => {
-                let is_java = args.first()?.ends_with("java");
-
-                if !is_java {
-                    None?
+                let is_java = args.first().map(String::as_str).unwrap_or_default().ends_with("java");
+
+                if is_java {
+                    args.iter()
+                        .find_map(|arg| arg.strip_prefix("-agentlib:jdwp=transport=dt_socket"))
+                        .and_then(|agent_lib_args| {
+                            agent_lib_args
+                                .split(',')
+                                .find_map(|arg| arg.strip_prefix("address="))
+                        })
+                        .and_then(|full_address| full_address.split(':').last())
+                        .and_then(|port| port.parse::<u16>().ok())
+                        .map(|port| SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), port))
+                } else {
+                    None
                 }
-
-                args.iter()
-                    .find_map(|arg| arg.strip_prefix("-agentlib:jdwp=transport=dt_socket"))
-                    .and_then(|agent_lib_args| agent_lib_args.split(',').find_map(|arg| arg.strip_prefix("address=")))
-                    .and_then(|full_address| full_address.split(':').last())
-                    .and_then(|port| port.parse::<u16>().ok())
-                    .map(|port| SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), port))
-
+                .into_iter()
+                .collect::<Vec<_>>()
             }
-        }
-        .and_then(|addr| match addr.ip() {
+            Self::NodeInspector => {
+                if let Some(value) = get_env("NODE_OPTIONS") {
+                    // matching specific flags so we avoid matching on, for example,
+                    // `--inspect-publish-uid=http`
+                    value.split_ascii_whitespace()
+                    .filter_map(|flag| match flag.split_once('=') {
+                        Some(("--inspect" | "--inspect-brk" | "--inspect-wait", port)) => port.parse::<u16>().ok(),
+                        None if ["--inspect", "--inspect-brk", "--inspect-wait"].contains(&flag) => Some(NODE_INSPECTOR_DEFAULT_PORT),
+                        _ => None,
+                    })
+                    .map(|port| SocketAddr::new(Ipv4Addr::LOCALHOST.into(), port))
+                    .collect::<Vec<_>>()
+                } else if let Some(value) = get_env("NODE_INSPECTOR_INFO") {
+                    value.split(',').filter_map(|var| match var.split_once(':')? {
+                        ("inspectorURL", url) => url.parse::<Uri>().ok()?.port_u16(),
+                        _ => None,
+                    })
+                    .map(|port| SocketAddr::new(Ipv4Addr::LOCALHOST.into(), port))
+                    .collect::<Vec<_>>()
```

Comment:
This env value is a json, right? We could use here sth like
```rust
#[derive(Deserialize)
#[serde(rename_all = "camelCase")]
struct NodeInspectorInfo {
    inspector_url: Uri,
}

serde_json::from_str::<NodeInspectorInfo>(&value)
    .inspect_err(|error| tracing::warn!(%error, value, "Failed to deserialize NODE_INSPECTOR_INFO"))
    .ok()
    .map(|uri| {
        vec![SocketAddr::new(Ipv4Addr::LOCALHOST.into(), uri.port_u16())]
    })
    .unwrap_or_default()
```

Manual parsing is prone to changes in the representation. Like, at some point they might add extra whitespace. It's highly unlikely, so leaving it up to you

Reply:
i like this but i think it might be overkill here? ill leave it out for now